frank-flow/src/frontend/src/app/flow/node/node.component.ts

```diff
@@ -6,126 +6,194 @@ import {
+      ? this.makeNodeListener(bottomEndpointOptions)
+      : this.makeNodeTarget(topEndpointOptions, id);
+    if (this.nodeIsNotExit()) {
+      this.makeNodeSource(bottomEndpointOptions, id);
+    }
+
+    this.jsPlumbInstance.draggable(id, this.getDragOptions());
+  }
+
+  nodeIsNotExit() {
+    return this.cssClass !== 'shape--round color--danger';
+  }
+
+  nodeIsListener() {
+    return this.cssClass === 'shape--oval color--info';
+  }
+
+  makeNodeListener(bottomEndpointOptions: EndpointOptions): void {
+    bottomEndpointOptions.isSource = false;
+    bottomEndpointOptions.connectionsDetachable = false;
+  }
+
+  makeNodeSource(bottomEndpointOptions: EndpointOptions, id: string): void {
+    this.jsPlumbInstance.addEndpoint(
+      id,
+      {
+        anchor: this.getSourceAnchor(),
+        uuid: id + '_bottom',
+        maxConnections: -1,
+      },
+      bottomEndpointOptions
+    );
+  }
+
+  makeNodeTarget(topEndpointOptions: EndpointOptions, id: string): void {
+    this.jsPlumbInstance.addEndpoint(
+      id,
+      { anchor: this.getTargetAnchor(), uuid: id + '_top', maxConnections: -1 },
+      topEndpointOptions
+    );
+  }
+
+  getSourceAnchor(): AnchorSpec {
+    return this.settings.verticalConnectors ? 'Bottom' : 'RightMiddle';
+  }
+
+  getTargetAnchor(): AnchorSpec {
+    return this.settings.verticalConnectors ? 'Top' : 'LeftMiddle';
+  }
+
+  getTopendpointOptions(): EndpointOptions {
+    return {
+      endpoint: ['Dot', { radius: 7 }],
+      paintStyle: { fill: '#ffcb3a' },
+      isSource: false,
+      scope: 'jsPlumb_DefaultScope',
+      maxConnections: 1,
+      isTarget: true,
+      dropOptions: this.getDropOptions(),
+    };
+  }
+
+  getBottomEndpointOptions(): EndpointOptions {
+    return {
+      endpoint: ['Dot', { radius: 10 }],
+      paintStyle: { fill: '#99cb3a' },
+      isSource: true,
+      scope: 'jsPlumb_DefaultScope',
+      connectorStyle: { stroke: '#99cb3a', strokeWidth: 3 },
+      connector: this.getConnector(),
+      maxConnections: 30,
+      isTarget: false,
+      connectorOverlays: [['Arrow', { location: 1 }]],
+      dropOptions: this.getDropOptions(),
+    };
+  }
+
+  getDragOptions(): any {
+    return {
+      tolerance: 'touch',
+      hoverClass: 'dropHover',
+      activeClass: 'dragActive',
+    };
+  }
+
+  getDropOptions(): any {
+    return {
+      containment: 'canvas',
+      grid: [20, 20],
+      stop: (e: any) => {
+        this.handleDragStop(e);
+      },
+    };
+  }
+
+  getConnector(): ConnectorSpec {
+    switch (+this.settings.connectionType) {
+      case ConnectionType.flowchart:
+        return ['Flowchart', { alwaysRespectStubs: true, cornerRadius: 25 }];
+      case ConnectionType.bezier:
+        return [
+          'Bezier',
+          {
+            alwaysRespectStubs: true,
+            cornerRadius: 10,
+            stub: [10, 50],
+            midpoint: 0.0001,
+          },
+        ];
+      default:
+        return ['Straight', {}];
```

Comment:
I think there is a very valid reason to use a default for when we eventually expand the `ConnectionType` enum. I agree that 'straight' should have a case. I have chosen to throw an error in the default so that we won't forget to expand the switch case for when the enum gets expanded.